### PR TITLE
Update Aggregate Attestation

### PR DESCRIPTION
**Pull request by nisdas**
This updates attestation service to accurately retrieve a validator from the aggregate bitfield in line with the spec.

### CODE REVIEW

File: beacon-chain/attestation/service.go [hunk 1]
```diff
@@ -5,6 +5,8 @@ import (
 	"context"
 	"fmt"
 
+	"github.com/prysmaticlabs/prysm/beacon-chain/core/helpers"
+
```
Comment (terenc3t) [inline beacon-chain/attestation/service.go]: spacing

File: beacon-chain/attestation/service.go [hunk 2]
```diff
@@ -174,6 +176,22 @@ func (a *Service) updateLatestAttestation(ctx context.Context, attestation *pb.A
...
 	}
+
+	var committee []uint64
+	// We find the crosslink committee for the shard and slot by the attestation.
+	committees, err := helpers.CrosslinkCommitteesAtSlot(state, attestation.Data.Slot, false /* registryChange */)
+	if err != nil {
+		return err
+	}
+
+	// Find committee for shard
```
Comment (terenc3t) [inline beacon-chain/attestation/service.go]:
```suggestion
	// Find committee for shard.
```

File: beacon-chain/attestation/service.go [hunk 3]
```diff
@@ -190,8 +208,10 @@ func (a *Service) updateLatestAttestation(ctx context.Context, attestation *pb.A
 		if !bitSet {
 			continue
 		}
-		// If the attestation came from this attester.
-		pubkey := bytesutil.ToBytes48(state.ValidatorRegistry[i].Pubkey)
+
+		// If the attestation came from this attester. We use the slot committee to find the
+		// validator's actual index
```
Comment (terenc3t) [inline beacon-chain/attestation/service.go]:
```suggestion
		// validator's actual index.
```

File: beacon-chain/attestation/service_test.go [hunk 4]
```diff
@@ -25,8 +25,17 @@ func TestUpdateLatestAttestation_UpdatesLatest(t *testing.T) {
 	defer internal.TeardownDB(t, beaconDB)
 	ctx := context.Background()
 
+	var validators []*pb.Validator
+	for i := 0; i < 64; i++ {
+		validators = append(validators, &pb.Validator{
+			Pubkey:          []byte{'A'},
```
Comment (terenc3t) [inline beacon-chain/attestation/service_test.go]: is it ok that all 64 validators have the same pub key?
 just to make sure it doesn't affect the test integrety

Comment (rauljordan) [reply beacon-chain/attestation/service_test.go]: I agree to use different ones - just in case this doesn't catch regression in future changes

Comment (nisdas) [reply beacon-chain/attestation/service_test.go]: yeah that was an oversight, will change it

File: beacon-chain/attestation/service.go (hunk 1 above)
Comment (prestonvanloon) [reply beacon-chain/attestation/service.go]: +1

**Pull request merged**